File: 2502-sort-the-people/2502-sort-the-people.cpp
#include <algorithm>
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        for(int i = 0; i < names.size() - 1; i++){
            for(int j = 0; j < names.size() - 1 -i;j++){
                if(heights[j+1] > heights[j]){

                    swap(heights[j+1], heights[j]);
                    swap(names[j+1], names[j]);
                }
            }
        }
        return names;
    }
};

File: 2502-sort-the-people/2502-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        #sorting using insertion sort
        
        zipped = list(zip(names,heights))
        zipped.sort(key = lambda x: x[1],reverse = True)
        return [name for name,height in zipped]